tests/components/yamaha/test_media_player.py

```diff
@@ -23,54 +26,142 @@ def __init__(self, ctrl_url, name, zones=None):
         """Initialize the fake Yamaha device."""
         self.ctrl_url = ctrl_url
         self.name = name
-        self.zones = zones or []
+        self._zones = zones or []
 
     def zone_controllers(self):
         """Return controllers for all available zones."""
-        return self.zones
+        return self._zones
+
+
+@pytest.fixture(name="main_zone")
+def main_zone_fixture():
+    """Mock the main zone."""
+    return _create_zone_mock("Main zone", "http://main")
+
+
+@pytest.fixture(name="device")
+def device_fixture(main_zone):
+    """Mock the yamaha device."""
+    device = FakeYamahaDevice("http://receiver", "Receiver", zones=[main_zone])
+    with patch("rxv.RXV", return_value=device):
+        yield device
+
+
+async def test_setup_host(hass, device, main_zone):
+    """Test set up integration with host."""
+    assert await async_setup_component(hass, mp.DOMAIN, CONFIG)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5662368)